.github/workflows/test-jinad.yml

```diff
@@ -0,0 +1,56 @@
+name: Test jinad
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  jinad-test:
+    if: |
+      !startsWith(github.event.head_commit.message, 'chore') &&
+      !startsWith(github.event.head_commit.message, 'build: hotfix') &&
+      !endsWith(github.event.head_commit.message, 'reformatted by jina-dev-bot')
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.7]
+        path: [jinad]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Get jinad repo
+        uses: actions/checkout@v2
+        with:
+          repository: jina-ai/jinad
+          path: jinad
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[cicd,test,http] requests --no-cache-dir
+      - name: Test example with pytest
+        run: |
+          jina check
+          export JINA_LOG_VERBOSITY="ERROR"
+          cd jinad/${{ matrix.path }}
+          pip install -r requirements.txt --no-cache-dir
+          pytest -s -v .
+        timeout-minutes: 30
+        env:
+          JINAHUB_USERNAME: ${{ secrets.JINAHUB_USERNAME }}
+          JINAHUB_PASSWORD: ${{ secrets.JINAHUB_PASSWORD }}
+      - name: Test example with pytest
+        run: |
+          cd jinad/${{ matrix.path }}
+          ./tests/integration/jinad/test_simple_distributed/test_integration.sh
```

Comment:
Not for now, but I would have a `.sh` in the root of `jinad` folder that similar to pytest, captures every test in the child folders and executes them, and reports the failing ones.

Reply:
This maps to the `jinad/jinad` folder (first directory `jinad` inside `jinad` repo)